paasta_tools/autoscaling_lib.py

```diff
@@ -753,16 +755,17 @@ def scale_aws_spot_fleet_request(resource, current_capacity, target_capacity, so
                 break
             # The start time of the maintenance window is the point at which
             # we giveup waiting for the instance to drain and mark it for termination anyway
-            start = int(time.time() + CLUSTER_DRAIN_TIMEOUT)
+            start = int(time.time() + CLUSTER_DRAIN_TIMEOUT) * 1000000000  # nanoseconds
             # Set the duration to an hour, if we haven't cleaned up and termintated by then
             # mesos should put the slave back into the pool
-            duration = 600
+            duration = 600 * 1000000000  # nanoseconds
             log.info("Draining {0}".format(slave_to_kill['pid']))
             if not dry_run:
                 try:
-                    drain([slave_to_kill['ip']], start, duration)
+                    drain([slave_to_kill['hostname']], start, duration)
```

Comment:
Draining actually requires specifying both a hostname and an IP address. As a convenience, I made paasta_maintenance attempt to query DNS to get the IP address, but I also left a way to bypass this. If you pass a hostname of the form `<hostname>|<ip>`, it will use the values you provide and skip the extra DNS lookup.

Since the autoscaling_lib is aware of both the IP and hostname it is trying to drain, it is probably best if it passes both to paasta_maintenance.